test/tests/functional/pbs_cray_hyperthread.py

```diff
@@ -0,0 +1,102 @@
+# coding: utf-8
+
+# Copyright (C) 1994-2018 Altair Engineering, Inc.
+# For more information, contact Altair at www.altair.com.
+#
+# This file is part of the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# PBS Pro is free software. You can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# For a copy of the commercial license terms and conditions,
+# go to: (http://www.pbspro.com/UserArea/agreement.html)
+# or contact the Altair Legal Department.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+from tests.functional import *
+from ptl.utils.pbs_crayutils import CrayUtils
+import os
+
+
+class TestCrayHyperthread(TestFunctional):
+
+    """
+    The test will submit a job script that call's aprun with the
+    option that will allow callers to use the hyperthreads
+    on a hyperthreaded compute node.
+    """
+
+    def setUp(self):
+        if not self.du.get_platform().startswith('cray'):
+            self.skipTest("Test suite only meant to run on a Cray")
+        TestFunctional.setUp(self)
+
+    def test_hyperthread(self):
+        """
+        Check for a compute node that has hyperthreads, if there is one
+        submit a job to that node requesting the hyperthreads.  Check
+        there are no errors in the job error output.
+        If there is no node with hyperthreads, skip the test.
+        """
+        # Get the compute nodes from PBS and see if they are threaded
+        cu = CrayUtils()
+        all_nodes = self.server.status(NODE)
+        self.logger.info("all nodes = %s" % all_nodes)
```

Comment:
do we need this log? It might be huge depending on how many compute nodes are there in the complex.

Reply:
Nope, it's a vestige from debugging.  Removed now.